linode/databasepostgresql/framework_models.go

```diff
@@ -0,0 +1,120 @@
+	Port                  types.Int64  `tfsdk:"port"`
+	RootPassword          types.String `tfsdk:"root_password"`
+	Status                types.String `tfsdk:"status"`
+	Updated               types.String `tfsdk:"updated"`
+	Updates               types.List   `tfsdk:"updates"`
+	RootUsername          types.String `tfsdk:"root_username"`
+	Version               types.String `tfsdk:"version"`
+	ID                    types.Int64  `tfsdk:"id"`
+}
+
+func (data *DataSourceModel) parsePostgresDatabase(
+	ctx context.Context, db *linodego.PostgresDatabase,
+) diag.Diagnostics {
+	data.DatabaseID = types.Int64Value(int64(db.ID))
+	data.Status = types.StringValue(string(db.Status))
+	data.Label = types.StringValue(string(db.Label))
+	data.HostPrimary = types.StringValue(string(db.Hosts.Primary))
+	data.HostSecondary = types.StringValue(string(db.Hosts.Secondary))
+	data.Region = types.StringValue(string(db.Region))
+	data.Type = types.StringValue(string(db.Type))
+	data.Engine = types.StringValue(string(db.Engine))
+	data.Port = types.Int64Value(int64(db.Port))
+	data.Version = types.StringValue(string(db.Version))
+	data.ClusterSize = types.Int64Value(int64(db.ClusterSize))
+	data.ReplicationType = types.StringValue(string(db.ReplicationType))
+	data.ReplicationCommitType = types.StringValue(string(db.ReplicationCommitType))
+	data.SSLConnection = types.BoolValue(db.SSLConnection)
+	data.Encrypted = types.BoolValue(db.Encrypted)
+
+	allowList, diags := types.SetValueFrom(ctx, types.StringType, db.AllowList)
+	if diags.HasError() {
+		return diags
+	}
+	data.AllowList = allowList
+
+	data.Created = types.StringValue(db.Created.Format(time.RFC3339))
+	data.Updated = types.StringValue(db.Updated.Format(time.RFC3339))
+
+	data.EngineID = types.StringValue(helper.CreateDatabaseEngineSlug(db.Engine, db.Version))
+
+	updates, diags := flattenDatabaseMaintenanceWindow(ctx, db.Updates)
+	if diags.HasError() {
+		return diags
+	}
+
+	data.Updates = *updates
+
+	data.ID = types.Int64Value(int64(db.ID))
+
+	return nil
+}
+
+func (data *DataSourceModel) parsePostgresDatabaseSSL(db *linodego.PostgresDatabaseSSL) {
+	data.CACert = types.StringValue(string(db.CACertificate))
+}
+
+func (data *DataSourceModel) parsePostgresDatabaseCredentials(db *linodego.PostgresDatabaseCredential) {
+	data.RootUsername = types.StringValue(db.Username)
+	data.RootPassword = types.StringValue(db.Password)
+}
+
+func flattenDatabaseMaintenanceWindow(ctx context.Context, maintenance linodego.DatabaseMaintenanceWindow) (
+	*basetypes.ListValue, diag.Diagnostics,
+) {
+	result := make(map[string]attr.Value)
+
+	result["day_of_week"] = types.StringValue(helper.FlattenDayOfWeek(maintenance.DayOfWeek))
+	result["duration"] = types.Int64Value(int64(maintenance.Duration))
+	result["frequency"] = types.StringValue(string(maintenance.Frequency))
+	result["hour_of_day"] = types.Int64Value(int64(maintenance.HourOfDay))
+	result["week_of_month"] = types.Int64Value(int64(*maintenance.WeekOfMonth))
```

Comment:
```suggestion
	if maintenance.WeekOfMonth != nil {
		result["week_of_month"] = types.Int64Value(int64(*maintenance.WeekOfMonth))
	} else {
		result["week_of_month"] = types.Int64Null()
	}
```

I think we can handle the nil to avoid null pointer issue and align with [the original behavior of this data source.](https://github.com/linode/terraform-provider-linode/blob/framework-migration-phase-1/linode/helper/database.go#L83-L86)

I am going to double check this but it is taking half hour to provision a db so I can't test it right now.

CC: @ezilber-akamai @yec-akamai for double checking if I was incorrect.

Reply:
Sounds correct to me. I remember @zliang-akamai you've built a helper function to handle the default value for int64 right? We can use it here, so when we decide to have a breaking change, it's easier to find all the places that set the default value.

Reply:
Fixed this! I also made the same change in the `databasemysql` framework models as it had the same problem.

Reply:
Just double checked it hasn't been merged: https://github.com/linode/terraform-provider-linode/pull/850/files#diff-5862907430fd194670e982db3fc2ad3b39f30401f6124f89e5a7263be8fbdadaR17 I think you can copy it from that PR.

Reply:
Oh I ended up writing the same thing XD